user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -291,6 +291,43 @@ def __process_filter_args(self, arg_val: str):
                 selected_filter = QualFilterApp.fromstring(default_filter_txt)
         self.ctxt.set_ctxt('filterApps', selected_filter)
 
+    def _process_price_discount_args(self):
+        raw_cpu_discount = self.wrapper_options.get('cpuDiscount')
+        raw_gpu_discount = self.wrapper_options.get('gpuDiscount')
+        raw_global_discount = self.wrapper_options.get('globalDiscount')
+        if raw_global_discount is not None and (raw_cpu_discount is not None or raw_gpu_discount is not None):
+            self.logger.error('Setting both global_discount and either cpu_discount or '
+                              'gpu_discount is inconsistent.')
+            raise RuntimeError('Invalid arguments. If global_discount is specified, no additional '
+                               'discount arguments (cpu_discount or gpu_discount) should be set.')
+        try:
+            cpu_discount = int(raw_cpu_discount) if raw_cpu_discount is not None else 0
+            gpu_discount = int(raw_gpu_discount) if raw_gpu_discount is not None else 0
+            global_discount = int(raw_global_discount) if raw_global_discount is not None else 0
+        except Exception as ex:
+            self.logger.error('Discount arguments have incorrect type.')
+            raise RuntimeError('Invalid arguments. Discount arguments cannot be converted to integer.') from ex
+
+        if cpu_discount < 0 or cpu_discount > 100:
```

Comment:
Nit: You could call a method to do this check which will remove the redundant code.
Something like:
```
def check_discount_precentage(discount_type, discount_value):
    if  discount_value < 0 or  discount_value > 100:
        self.logger.error(f' {discount_type} is out of range [0, 100]')
            raise RuntimeError(f'Invalid arguments. {discount_type} = {discount_value} is an invalid '
                               'percentage.')

```

Reply:
Thanks! I will update this.